Microsoft.Toolkit.Uwp.UI.Controls.Graph/PeoplePicker/PeoplePicker.cs

```diff
@@ -40,28 +59,231 @@ protected override void OnApplyTemplate()
 
-            SearchResultList = new ObservableCollection<Person>();
-            Selections = Selections ?? new ObservableCollection<Person>();
             if (_searchBox != null)
             {
-                _searchBox.TextChanged += SearchBox_OnTextChanged;
+                _searchBox.SizeChanged += SearchBox_OnSizeChanged;
+                _searchBox.GotFocus += SearchBox_GotFocus;
+            }
+
+            if (_internalSearchBox != null)
+            {
+                _internalSearchBox.GotFocus += SearchBox_GotFocus;
             }
 
             if (_searchResultListBox != null)
             {
-                _searchResultListBox.SelectionChanged += SearchResultListBox_OnSelectionChanged;
+                _searchResultListBox.Tapped += SearchResultListBox_Tapped;
+                _searchResultListBox.KeyUp += SearchResultListBox_KeyUp;
             }
 
             if (_selectionsListBox != null)
             {
                 _selectionsListBox.Tapped += SelectionsListBox_Tapped;
+                if (!IsWindowsPhone)
+                {
+                    _selectionsListBox.PreviewKeyUp += SelectionsListBox_KeyUp;
+                }
             }
 
             base.OnApplyTemplate();
         }
+
+        private void ClearAndHideSearchResultListBox()
+        {
+            SearchResults.Clear();
+            HideSearchResults();
+        }
+
+        private void ShowSearchResults()
+        {
+            FlyoutBase.ShowAttachedFlyout(_flyoutContainer);
+            _searchBox.Opacity = 0;
+        }
+
+        private void HideSearchResults()
+        {
+            FlyoutBase.GetAttachedFlyout(_flyoutContainer)?.Hide();
+        }
+
+        private void RaiseSelectionChanged()
+        {
+            SelectionChanged?.Invoke(this, new PeopleSelectionChangedEventArgs(Selections));
+        }
+
+        private Person GetPersonFromUser(User user)
+        {
+            Person person = new Person
+            {
+                Id = user.Id,
+                DisplayName = user.DisplayName,
+                GivenName = user.GivenName,
+                Surname = user.Surname,
+                JobTitle = user.JobTitle,
+                OfficeLocation = user.OfficeLocation,
+                UserPrincipalName = user.UserPrincipalName,
+                ScoredEmailAddresses = new ScoredEmailAddress[] { new ScoredEmailAddress() { Address = user.Mail } },
+                AdditionalData = new Dictionary<string, object>() { [EtagHelper.ODataEtagPropertyName] = user.GetEtag() }
+            };
+            return person;
+        }
+
+        private async Task SearchPeopleAsync(string searchPattern)
+        {
+            if (string.IsNullOrWhiteSpace(searchPattern))
+            {
+                ClearAndHideSearchResultListBox();
+                return;
+            }
+
+            IsLoading = true;
+            try
+            {
+                MicrosoftGraphService graphService = MicrosoftGraphService.Instance;
```

Comment:
I see the following lines repeated over and over
```
MicrosoftGraphService graphService = MicrosoftGraphService.Instance;
await graphService.TryLoginAsync();
```
Likewise for the following
```
MicrosoftGraphService graphService = MicrosoftGraphService.Instance;
await graphService.TryLoginAsync();
GraphServiceClient graphClient = graphService.GraphProvider;
```

How about creating some handy methods for this

```
internal static async Task<IGraphService> GetGraphService()
{
    MicrosoftGraphService graphService = MicrosoftGraphService.Instance;
    await graphService.TryLoginAsync();
    return graphService;
}

internal static async Task<IGraphServiceClient> GetGraphServiceClient()
{
    var graphService = await GetGraphService();
    return graphService.GraphProvider;
}

Reply:
@skendrot updated as per your advice